acceptance/helpers/auth/dex.go

```diff
@@ -0,0 +1,76 @@
+package auth
+
+import (
+	"encoding/base64"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/epinio/epinio/acceptance/helpers/proc"
+	"github.com/epinio/epinio/acceptance/testenv"
+	"github.com/epinio/epinio/helpers"
+
+	. "github.com/onsi/gomega"
+)
+
+type User struct {
+	username   string
+	password   string
+	role       string
+	workspaces []string
+}
+
+var users = []User{
+	{"admin@epinio.io", "password", "admin", []string{""}},
+	{"user1@epinio.io", "password", "user", []string{""}},
+	{"user2@epinio.io", "password", "user", []string{""}},
+}
+
+func InitUsers(env *testenv.EpinioEnv, apiURL string) {
+	createDexUsers(apiURL)
+
+	restartDex()
+
+	createEpinioUsersAndLogin(env, apiURL)
+}
+
+func createDexUsers(apiURL string) {
+	domain := strings.TrimPrefix(apiURL, "https://epinio.")
+
+	// get clientSecret
+	out, err := proc.Kubectl("get", "secret", "-n", "epinio", "dex-client-secret", "--template={{.data.clientSecret}}")
+	Expect(err).ToNot(HaveOccurred())
+	clientSecret, err := base64.StdEncoding.DecodeString(out)
+	Expect(err).ToNot(HaveOccurred())
+
+	dexConfSecretTemplate, err := os.ReadFile(testenv.TestAssetPath("dex-conf.yaml"))
+	Expect(err).ToNot(HaveOccurred())
+
+	dexConf := fmt.Sprintf(string(dexConfSecretTemplate), domain, clientSecret)
+	filePath, err := helpers.CreateTmpFile(dexConf)
+	Expect(err).ToNot(HaveOccurred())
+
+	// apply conf
+	out, err = proc.Kubectl("apply", "-f", filePath)
+	Expect(err).ToNot(HaveOccurred(), out)
+}
+
+func restartDex() {
+	// sometimes this fails for empty patch
+	_, _ = proc.Kubectl("rollout", "restart", "-n", "epinio", "deployment/dex")
+	out, err := proc.Kubectl("rollout", "status", "-n", "epinio", "deployment/dex")
+	Expect(err).ToNot(HaveOccurred(), out)
+
+	time.Sleep(10 * time.Second)
```

Comment:
We are waiting for Dex to restart and to "stabilize" (getting the conf and get ready).
This was done because we were adding some users to do the tests, but with the latest helm-charts we actually added them in the configuration, so it's possible to authenticate without actually "hack" the configuration and restart it.
I think I can drop this part all together. :smile: